### PR TITLE
Switch to go.mod

### PR DESCRIPTION
**Pull request by soltysh**
/assign @tnozicka 
since you asked for it :wink: 

Needs https://github.com/openshift/release/pull/5313 to merge

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,32 +37,23 @@ $(call build-image,ocp-cluster-kube-controller-manager-operator,$(IMAGE_REGISTRY
...
-.PHONY: verify-codegen-crds
 verify-codegen: verify-codegen-crds
 .PHONY: verify-codegen
 
-verify: verify-codegen
+test-e2e: GO_TEST_PACKAGES :=./test/e2e/...
+test-e2e: test-unit
+.PHONY: test-e2e
+
+clean:
```
Comment (tnozicka) [inline Makefile]: clean is automatic btw. now (but preexisting)

File: manifests/0000_25_kube-controller-manager-operator_01_config.crd.yaml [hunk 2]
```diff
@@ -30,216 +29,6 @@ spec:
             submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
           type: string
         metadata:
-          description: ObjectMeta is metadata that all persisted resources must have,
```
Comment (tnozicka) [inline manifests/0000_25_kube-controller-manager-operator_01_config.crd.yaml]: why it was generating ObjectMeta before?

**Comment by tnozicka** [comment]
lgtm % crd question

File: manifests/0000_25_kube-controller-manager-operator_01_config.crd.yaml (hunk 2 above)
Comment (soltysh) [reply manifests/0000_25_kube-controller-manager-operator_01_config.crd.yaml]: I think it's caused by older generator, but @damemi will know better.

**Comment by soltysh** [comment]
/override verify-deps
since that is not required since https://github.com/openshift/release/pull/5313

**Pull request merged**